### PR TITLE
Add "_id" to ES vector config; enable ES vector tests

### PR DESCRIPTION
**Pull request by vimalk78**
### Description
* Enabled Vector in Dockerfile and CSV
* Added "_id" to elasticsearch vector config
* Added Vector component type to e2e test framework, and made one elasticsearch test case to run with both fluentd and vector collector

/cc @jcantrill @ajaygupta978 
/assign

### CODE REVIEW

File: test/framework/functional/vector/deploy.go
Comment (jcantrill) [inline test/framework/functional/vector/deploy.go:1]: Is this cruft some a previous PR?

**Review by jcantrill — COMMENTED** [review]
/hold
/approve

File: test/framework/functional/vector/deploy.go
Comment (vimalk78) [reply test/framework/functional/vector/deploy.go:1]: yes, i was working on vector functional test earlier. the file got added by mistake

Comment (jcantrill) [reply test/framework/functional/vector/deploy.go:1]: Please remove before merge

Comment (vimalk78) [reply test/framework/functional/vector/deploy.go:1]: removed the file

Comment (vimalk78) [reply test/framework/functional/vector/deploy.go:1]: ?? i pushed a commit with file deleted

Comment (vimalk78) [reply test/framework/functional/vector/deploy.go:1]: i dont understand, i deleted this file, and this file does not exist when i have staged the files.
But when i create the commit, this file gets created
```
vimalkum (vector-es-add-id +) cluster-logging-operator $ git commit -m "Add "_id" to ES vector config; enable ES vector tests"
Dockerfile  << cannot modify Dockerfile
Dockerfile.in
bundle/manifests/cluster-logging.v5.4.0.clusterserviceversion.yaml
internal/generator/vector/conf_test.go
internal/generator/vector/output/elasticsearch/elasticsearch.go
internal/generator/vector/output/elasticsearch/elasticsearch_test.go
internal/generator/vector/output/elasticsearch/output_conf_es_test.go
internal/generator/vector/pipelines_to_output.go
manifests/5.4/cluster-logging.v5.4.0.clusterserviceversion.yaml
test/e2e/logforwarding/elasticsearchmanaged/clo_managed_logstore_test.go
test/e2e/logforwarding/elasticsearchmanaged/forwarding_with_defaults_test.go
test/framework/e2e/framework.go
test/helpers/clusterlogging.go
[vector-es-add-id 9a284379] Add _id to ES vector config; enable ES vector tests
 15 files changed, 278 insertions(+), 353 deletions(-)
 create mode 100644 test/framework/functional/vector/deploy.go
 delete mode 100644 test/helpers.go
```

**Review by ajaygupta978 — APPROVED** [review]

File: test/framework/functional/vector/deploy.go
Comment (vimalk78) [reply test/framework/functional/vector/deploy.go:1]: the files does not exist in local also
```
$ git rm test/framework/functional/vector/deploy.go
fatal: pathspec 'test/framework/functional/vector/deploy.go' did not match any files
$ rm test/framework/functional/vector/deploy.go
rm: cannot remove 'test/framework/functional/vector/deploy.go': No such file or directory
```

Comment (vimalk78) [reply test/framework/functional/vector/deploy.go:1]: may be my git index is corrupted?

Comment (vimalk78) [reply test/framework/functional/vector/deploy.go]: managed to fix it by moving changed to new branch and renaming new branch back to this one.

Please lgtm

**Pull request merged**
